test/IRGen/unmanaged_objc_throw_func.swift

```diff
@@ -0,0 +1,70 @@
+// RUN: %target-swift-frontend -emit-ir %s | %FileCheck %s
+
+import Foundation
+
+@objc protocol SR_9035_P {
+  func returnUnmanagedCFArray() throws -> Unmanaged<CFArray>
+}
+
+// CHECK: define hidden swiftcc %TSo10CFArrayRefa* @"$s25unmanaged_objc_throw_func9SR_9035_CC22returnUnmanagedCFArrays0G0VySo0H3RefaGyKF"(%T25unmanaged_objc_throw_func9SR_9035_CC* swiftself, %swift.error** noalias nocapture swifterror dereferenceable(8)) #0 {
+@objc class SR_9035_C: NSObject, SR_9035_P {
+  func returnUnmanagedCFArray() throws -> Unmanaged<CFArray> {
+    // CHECK: %3 = call swiftcc { %swift.bridge*, i8* } @"$ss27_allocateUninitializedArrayySayxG_BptBwlF"(i64 1, %swift.type* @"$sSiN")
```

Comment:
The SSA value numbers (`%3` et al) are pretty closely tied to the particular IR that gets emitted today. To avoid that, you can use a variable pattern: `%[[ALLOCATION:.+]] = call …`. Then in later expressions, you can just use `%[[ALLOCATION]]` to refer to `%3`, and that way it'll be easier to update if an instruction is added or deleted later.

(These features are all in https://llvm.org/docs/CommandGuide/FileCheck.html if you want to take a look later.)